packages/rspack/src/builtin-plugin/lazy-compilation/middleware.ts

```diff
@@ -36,20 +36,40 @@ const getFullServerUrl = ({ serverUrl, prefix }: LazyCompilationOptions) => {
 	);
 };
 
+const DEPRECATED_LAZY_COMPILATION_OPTIONS_WARN =
+	"The `experiments.lazyCompilation` option is deprecated, please use the configuration top level `lazyCompilation` instead.";
+
 export const lazyCompilationMiddleware = (
 	compiler: Compiler | MultiCompiler
 ): MiddlewareHandler => {
 	if (compiler instanceof MultiCompiler) {
 		const middlewareByCompiler: Map<string, MiddlewareHandler> = new Map();
 
 		let i = 0;
+		let isReportDeprecatedWarned = false;
 		for (const c of compiler.compilers) {
-			if (!c.options.experiments.lazyCompilation) {
+			if (c.options.experiments.lazyCompilation) {
+				if (c.name) {
+					console.warn(
+						`The 'experiments.lazyCompilation' option in compiler named '${c.name}' is deprecated, please use the Configuration top level 'lazyCompilation' instead.`
+					);
+				} else if (!isReportDeprecatedWarned) {
+					console.warn(DEPRECATED_LAZY_COMPILATION_OPTIONS_WARN);
+					isReportDeprecatedWarned = true;
+				}
+			}
+
+			if (
+				!c.options.lazyCompilation &&
+				!c.options.experiments.lazyCompilation
+			) {
 				continue;
 			}
 
 			const options = {
-				...c.options.experiments.lazyCompilation
+				// TODO: remove this when experiments.lazyCompilation is removed
+				...c.options.experiments.lazyCompilation,
```

Comment:
we may need add a warning when setting experiments.lazyCompilation and options.lazyCompilation together and tell users that options.lazyCompilation override experiments.lazyCompilation

Reply:
Updated